app/models/terms_of_service.rb

```diff
@@ -6,29 +6,50 @@
 #
 #  id                   :bigint(8)        not null, primary key
 #  changelog            :text             default(""), not null
+#  effective_date       :date
 #  notification_sent_at :datetime
 #  published_at         :datetime
 #  text                 :text             default(""), not null
 #  created_at           :datetime         not null
 #  updated_at           :datetime         not null
 #
 class TermsOfService < ApplicationRecord
-  scope :published, -> { where.not(published_at: nil).order(published_at: :desc) }
-  scope :live, -> { published.limit(1) }
+  scope :published, -> { where.not(published_at: nil).order(effective_date: :desc) }
+  scope :live, -> { published.where('effective_date < now()').limit(1) }
   scope :draft, -> { where(published_at: nil).order(id: :desc).limit(1) }
 
   validates :text, presence: true
-  validates :changelog, presence: true, if: -> { published? }
+  validates :changelog, :effective_date, presence: true, if: -> { published? }
+
+  validate :effective_date_cannot_be_in_the_past
 
   def published?
     published_at.present?
   end
 
+  def effective?
+    published? && effective_date.past?
```

Comment:
`effective_date` can be `nil`, for instance for ToS entries created before this PR. In such cases, this will raise `NoMethodError (undefined method 'past?' for nil)`.